language/tools/move-mv-llvm-compiler/src/main.rs

```diff
@@ -103,18 +108,28 @@ fn main() -> anyhow::Result<()> {
                 // but need to check for possible reassignment, so making this in old fashion loop:
                 for (symbol, account_address) in account_addresses {
                     let name = symbol.as_str().to_string();
-                    let address =
-                        NumericalAddress::parse_str(&account_address.to_string()).unwrap();
-                    if let Some(value) = named_address_map.get(&name) {
-                        if *value != address {
-                            bail!("{} already has assigned address {}, cannot reassign with new address {}. Possibly an error in Move.toml.",
+                    let address_string = account_address.to_string();
+                    let address_string_hex = format!("0x{}", address_string);
+                    match NumericalAddress::parse_str(&address_string_hex).map_err(|s| {
+                        anyhow!(
+                            "Failed to parse numerical address '{}'. Got error: {}",
+                            address_string_hex,
+                            s
+                        )
+                    }) {
+                        Ok(address) => {
+                            if let Some(value) = named_address_map.get(&name) {
+                                if *value != address {
+                                    bail!("{} already has assigned address {}, cannot reassign with new address {}. Possibly an error in Move.toml.",
                             name, address, *value);
+                                }
+                            } else {
+                                named_address_map.insert(name, address);
+                            }
                         }
-                    } else {
-                        named_address_map.insert(name, address);
-                    }
+                        Err(err) => eprintln!("Error reding address: {}", err),
+                    };
```

Comment:
I think this whole block is equivalent to this code, which is shorter and more readable.

```suggestion
                    if let Some(value) = named_address_map.get(&name) {
                        let address_string_hex = format!("0x{}", account_address);
                        if *value
                            != NumericalAddress::parse_str(&address_string_hex).unwrap_or_else(
                                |err| {
                                    eprintln!(
                                        "Failed to parse numerical address '{}'. Got error: {}",
                                        address_string_hex, err
                                    );
                                    *value
                                },
                            )
                        {
                            bail!("{} already has assigned address {}, cannot reassign with new address {}. Possibly an error in Move.toml.",
                                  name, *value, address_string_hex);
                        }
                    }

```

Reply:
Yes, it is simple. Thx!